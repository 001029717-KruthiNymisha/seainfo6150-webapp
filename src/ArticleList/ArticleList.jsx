import React from "react";
import ArticleListItem from "../ArticleListItem/ArticleListItem";
import styles from "./ArticleList.module.css";

const ArticleList = (props) => {
  let displayContent;

  if (props.articles.length) {
    displayContent = (

      <ul className={styles.container}>
        {props.articles.map((item) => (
          <ArticleListItem article={item} />
        ))}
      </ul>

    );
  } else {
    displayContent = <div>You have no data!</div>;
  }
  return (
    <section>
      <header>
        <h2></h2>
      </header>
      {displayContent}
    </section>
  );
};

export default ArticleList;
