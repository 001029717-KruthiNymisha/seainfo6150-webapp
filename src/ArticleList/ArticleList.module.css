.container {
    display: grid;
    background: gray;
    grid-template-columns: repeat(3,1fr);
    grid-gap: 5px 33px;
}


