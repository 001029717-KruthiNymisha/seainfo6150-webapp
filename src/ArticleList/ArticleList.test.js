import React from "react";
import ArticleList from "./ArticleList";
import { fireEvent } from "@testing-library/react";

describe("ArticleListTests", () => {
    const articles = [
        {
            slug: "181122102925-statue-of-liberty-crown",
            title: "The Statue of Liberty's torch heads to new museum",
            shortText:
                "The Statue of Liberty's original torch is getting a new home",
            displayDate: 'November 22nd 2018, 7:12 am',
        },
        {
            slug: "170321135754-airport-security-line",
            title: 'You can take a pie on a plane',
            shortText:
                "Getting excited to eat your mother's roast turkey, your uncle's stuffing and your sister's famous green bean casserole?",
            displayDate: 'November 21st 2018, 2:57 pm',
        },
    ];
    it("renders correctly", () => {
        const { container } = render(<ArticleList articles={articles} />);
        expect(container).toMatchSnapshot();
    });
});