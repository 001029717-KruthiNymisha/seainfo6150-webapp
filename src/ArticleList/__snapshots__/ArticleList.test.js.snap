// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListTest renders correctly 1`] = `
<div>
  <section>
    <header>
      <h2 />
    </header>
    <ul
      class="container"
    >
      <li
        class="container"
      >
        <article
          class="article"
        >
          <div>
            <img
              class="image"
              src="https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg"
            />
            <h1
              class="title"
              src="The Statue of Liberty's torch heads to new museum"
            />
          </div>
          <h1
            class="title"
          >
            The Statue of Liberty's torch heads to new museum
          </h1>
          <button
            class="button"
          >
            Show more
          </button>
        </article>
      </li>
      <li
        class="container"
      >
        <article
          class="article"
        >
          <div>
            <img
              class="image"
              src="https://cdn.cnn.com/cnnnext/dam/assets/170321135754-airport-security-line-file-super-169.jpg"
            />
            <h1
              class="title"
              src="You can take a pie on a plane"
            />
          </div>
          <h1
            class="title"
          >
            You can take a pie on a plane
          </h1>
          <button
            class="button"
          >
            Show more
          </button>
        </article>
      </li>
    </ul>
  </section>
</div>
`;
